src/credentials.rs

```diff
@@ -32,54 +38,78 @@ pub struct Credentials {
 }
 
 impl Credentials {
-    fn location() -> eyre::Result<PathBuf> {
-        let home = home::home_dir().wrap_err("Failed to locate home directory")?;
+    /// Get buffrs config directory.
+    fn buffrs_home() -> Result<PathBuf, VarError> {
+        var("BUFFRS_HOME").map(PathBuf::from)
+    }
 
-        let home = env::var("BUFFRS_HOME").map(PathBuf::from).unwrap_or(home);
+    /// XDG path: uses whichever config directory is appropriate for the platform.
+    ///
+    /// For example, on Linux this might be `~/.config/buffrs/credentials.toml`.
+    fn xdg_path() -> PathBuf {
+        Self::buffrs_home()
+            .unwrap_or_else(|_| dirs::config_dir().expect("get config dir").join(BUFFRS_DIR))
+            .join(CREDENTIALS_FILE)
+    }
 
-        Ok(home.join(BUFFRS_HOME).join(CREDENTIALS_FILE))
+    /// Legacy path: hard-coded to `~/.buffrs/credentials.toml`.
+    fn legacy_path() -> PathBuf {
+        Self::buffrs_home()
+            .unwrap_or_else(|_| dirs::home_dir().expect("get home dir").join(BUFFRS_HOME))
+            .join(CREDENTIALS_FILE)
+    }
+
+    /// Possible locations for the credentials file
+    fn possible_locations() -> Vec<PathBuf> {
+        vec![Self::xdg_path(), Self::legacy_path()]
     }
 
     /// Checks if the credentials exists
     pub async fn exists() -> eyre::Result<bool> {
-        fs::try_exists(Self::location()?)
-            .await
-            .wrap_err("Failed to detect credentials")
+        for location in &Self::possible_locations() {
```

Comment:
I think this may be cleaner in functional style: `Self::possible_locations().iter().map(...).all()`.